src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,105 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
+
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private int currentSize;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        values[currentSize++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAddAndRemove(index);
+        ensureCapacity();
+        System.arraycopy(values, index, values, index + 1, currentSize - index);
+        values[index] = value;
+        currentSize++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexForGetAndSet(index);
+        return values[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexForGetAndSet(index);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexForAddAndRemove(index);
```

Comment:
why are you doing the check for `add()` and not the check for `set() and get()`?
is it possible to remove the element from index 4 from the array of size 4?